src/dataStores/models/dataStoresManager.ts

```diff
@@ -50,54 +50,53 @@ export class DataStoresManager {
-    this.logger.info({ msg: 'creating new dataStore', metadata: { workspaceName, createDataStoreBody } });
+  public async createDataStore(workspaceName: string, createDataStoreBody: DataStoreBodyRequest): Promise<void> {
+    this.logger.info({
+      msg: `creating new dataStore named:${createDataStoreBody.name} in workspace: ${workspaceName}`,
+      workspaceName,
+      createDataStoreBody,
+    });
     const dataStoreName = createDataStoreBody.name;
     //test if workspace exists
     await this.checkWorkspace(workspaceName);
     //test conflict
-    const doesDataStoreExist = await this.checkConflict(workspaceName, dataStoreName);
-    if (!doesDataStoreExist) {
-      const convertedCreateDataStoreBody = postDataStoreRequestConverter(createDataStoreBody, this.dataStoreConnectionParams);
-      await this.geoserverManager.postRequest<GeoServerCreateDataStoreRequest>(
-        `workspaces/${workspaceName}/datastores`,
-        convertedCreateDataStoreBody
-      );
-      //check that data store was really created, otherwise, delete it
-      //await this.validateCreationOfDataStore(workspaceName, dataStoreName);
-    } else {
+    const doesDataStoreExist = await this.checkDataStore(workspaceName, dataStoreName);
+    if (doesDataStoreExist) {
       const errorMessage = `Cant create new dataStore in ${workspaceName} named ${dataStoreName}, there is already a dataStore under this name`;
       this.logger.error({ msg: errorMessage });
       throw new ConflictError(errorMessage);
     }
+    const convertedCreateDataStoreBody = postDataStoreRequestConverter(createDataStoreBody, this.connectionParams);
+    await this.geoserverManager.postRequest<GeoServerCreateDataStoreRequest>(`workspaces/${workspaceName}/datastores`, convertedCreateDataStoreBody);
   }
 
   @withSpanAsyncV4
-  public async updateDataStore(workspaceName: string, dataStoreName: string, updateRequest: DataStoreNameRequest): Promise<void> {
-    this.logger.info({ msg: 'updating dataStore', workspaceName, dataStoreName, updateRequest });
-    const doesNewDataStoreExist = await this.checkConflict(workspaceName, updateRequest.name);
-    if (!doesNewDataStoreExist) {
-      const updatedDataStore = updateDataStoreRequestConverter(updateRequest);
-      await this.geoserverManager.putRequest<GeoServerUpdateDataStoreRequest>(
-        `workspaces/${workspaceName}/datastores/${dataStoreName}`,
-        updatedDataStore
-      );
-    } else {
+  public async updateDataStore(workspaceName: string, dataStoreName: string, updateRequest: DataStoreBodyRequest): Promise<void> {
+    this.logger.info({ msg: `updating dataStore: ${dataStoreName} from workspace: ${workspaceName}`, workspaceName, dataStoreName, updateRequest });
+    //test if workspace exists
+    await this.checkWorkspace(workspaceName);
+    const doesNewDataStoreExist = await this.checkDataStore(workspaceName, updateRequest.name);
```

Comment:
i would recommended `dataStoreExists` instead
same for workspace if u named it the same

Reply:
or `isDataStoreExists`